sdk/appconfiguration/azure-data-appconfiguration/src/samples/java/com/azure/data/appconfiguration/ConfigurationClientJavaDocCodeSnippets.java

```diff
@@ -297,21 +304,27 @@ public void listSettings() {
      */
     public void listSettingRevisions() {
         ConfigurationClient configurationClient = createSyncConfigurationClient();
-        // BEGIN: com.azure.data.applicationconfig.configurationclient.listSettingRevisions#SettingSelector
-        for (ConfigurationSetting revision : configurationClient.listSettingRevisions(new SettingSelector()
-            .keys("prodDBConnection"))) {
-            System.out.printf("Key: %s, Value: %s", revision.key(), revision.value());
-        }
-        // END: com.azure.data.applicationconfig.configurationclient.listSettingRevisions#SettingSelector
+        // BEGIN: com.azure.data.applicationconfig.configurationclient.listSettingRevisions#settingSelector
+        SettingSelector settingSelector = new SettingSelector().keys("prodDBConnection");
+        PagedIterable<ConfigurationSetting> csStream =  configurationClient.listSettingRevisions(settingSelector);
+        csStream.streamByPage().forEach(setting -> {
+            System.out.printf("Key: %s, Value: %s", setting.value().get(0).key(), setting.value().get(0).value());
+        });
+        // END: com.azure.data.applicationconfig.configurationclient.listSettingRevisions#settingSelector
+    }
 
-        /**
-         * Generates code sample for using {@link ConfigurationClient#listSettingRevisions(SettingSelector, Context)}
-         */
+    /**
+     * Generates code sample for using {@link ConfigurationClient#listSettingRevisions(SettingSelector, Context)}
+     */
+    public void listSettingRevisionsContext() {
+        ConfigurationClient configurationClient = createSyncConfigurationClient();
         // BEGIN: com.azure.data.applicationconfig.configurationclient.listSettingRevisions#SettingSelector-Context
-        for (ConfigurationSetting revision : configurationClient.listSettingRevisions(new SettingSelector()
-            .keys("prodDBConnection"), new Context(key2, value2))) {
-            System.out.printf("Key: %s, Value: %s", revision.key(), revision.value());
-        }
+        SettingSelector settingSelector = new SettingSelector().keys("prodDBConnection");
+        Context ctx = new Context(key2, value2);
+        PagedIterable<ConfigurationSetting> csStream =  configurationClient.listSettingRevisions(settingSelector, ctx);
+        csStream.streamByPage().forEach(setting -> {
+            System.out.printf("Key: %s, Value: %s", setting.value().get(0).key(), setting.value().get(0).value());
```

Comment:
Is this a typical use-case? `setting.value().get(0).key()`?

Reply:
I am following convention, Test annotation are defined in parent class ConfigurationClientTestBase.